js/libs/form.js

```diff
@@ -114,31 +114,67 @@ Fliplet.Widget.instance('form-builder', function(data) {
 
     if (fields.length && (data.saveProgress && typeof progress === 'object') || entry) {
       fields.forEach(function(field) {
-        if (entry && entry.data && typeof entry.data[field.name] !== 'undefined' && field.populateOnUpdate !== false) {
-          if (field._type === "flDate") {
-            if (Fliplet.Env.get('platform') === 'web') {
-              field.value = moment(entry.data[field.name]).format('DD MMMM YYYY');
-            } else {
-              field.value = moment(entry.data[field.name]).format('YYYY-MM-DD');
-            }
-          } else if (field._type === "flCheckbox" && !Array.isArray(entry.data[field.name])) {
-            field.value = [];
-          } else if (field._type === 'flImage') {
-            var img = entry.data[field.name];
-            field.value = [];
-            if (Array.isArray(img)) {
-              field.value = img;
-            } else if (typeof img === 'string') {
-              field.value.push(img);
-            }
+        if (entry && entry.data && field.populateOnUpdate !== false) {
```

Comment:
any reason on why `typeof entry.data[field.name] !== 'undefined'` has been removed?

Reply:
@squallstar it wasn't removed just moved to the 140th line.
I moved it because of [this](https://github.com/Fliplet/fliplet-studio/issues/5275) task. There we needed that when undefined was sent as data for the date field we should set the date field to the current date. And when we had that condition at the beginning we were passing this data. The same was needed for [this](https://github.com/Fliplet/fliplet-studio/issues/5276) task as well.